server/sonar-ce/src/main/java/org/sonar/ce/cluster/HazelcastClientWrapper.java

```diff
@@ -0,0 +1,106 @@
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.ce.cluster;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.core.HazelcastInstance;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.picocontainer.Startable;
+import org.sonar.api.config.Settings;
+import org.sonar.process.ProcessProperties;
+
+import static com.google.common.base.Preconditions.checkState;
+import static org.apache.commons.lang.StringUtils.isNotEmpty;
+import static org.sonar.process.cluster.ClusterConstants.CLIENT_UUIDS;
+
+/**
+ * This class will connect as a Hazelcast client to the local instance of Hazelcluster
+ */
+public class HazelcastClientWrapper implements Startable {
+
+  private final ClientConfig hzConfig;
+
+  private HazelcastInstance hzInstance;
+
+  public HazelcastClientWrapper(Settings settings) {
+    boolean clusterEnabled = settings.getBoolean(ProcessProperties.CLUSTER_ENABLED);
+    String clusterName = settings.getString(ProcessProperties.CLUSTER_NAME);
+    String clusterLocalEndPoint = settings.getString(ProcessProperties.CLUSTER_LOCALENDPOINT);
+
+    checkState(clusterEnabled, "Cluster is not enabled");
+    checkState(isNotEmpty(clusterLocalEndPoint), "LocalEndPoint have not been set");
+    checkState(isNotEmpty(clusterName), "sonar.cluster.name is missing");
+
+    hzConfig = new ClientConfig();
+    hzConfig.getGroupConfig().setName(clusterName);
+    hzConfig.getNetworkConfig().addAddress(clusterLocalEndPoint);
+
+    // Tweak HazelCast configuration
+    hzConfig
+      // Increase the number of tries
+      .setProperty("hazelcast.tcp.join.port.try.count", "10")
+      // Don't bind on all interfaces
+      .setProperty("hazelcast.socket.bind.any", "false")
+      // Don't phone home
+      .setProperty("hazelcast.phone.home.enabled", "false")
+      // Use slf4j for logging
+      .setProperty("hazelcast.logging.type", "slf4j");
+  }
+
+  public <E> Set<E> getSet(String name) {
+    return hzInstance.getSet(name);
+  }
+
+  public <E> List<E> getList(String name) {
+    return hzInstance.getList(name);
+  }
+
+  public <K, V> Map<K, V> getMap(String name) {
+    return hzInstance.getMap(name);
+  }
+
+  public <K,V> Map<K,V> getReplicatedMap(String name) {
+    return hzInstance.getReplicatedMap(name);
+  }
+
+  public String getClientUUID() {
+    return hzInstance.getLocalEndpoint().getUuid();
```

Comment:
there is something fishy with this hazelcast and uuid thing, let's discuss it offline